tsdb/db.go

```diff
@@ -824,6 +825,16 @@ func (db *DB) Compact() (returnErr error) {
 		}
 		// Consider only successful compactions for WAL truncation.
 		lastBlockMaxt = maxt
+		compactionDuration := time.Since(start)
+
+		if compactionDuration.Milliseconds() > maxt-mint {
+			level.Warn(db.logger).Log(
+				"msg", "compaction took longer than the block time range, compactions are falling behind and won't be able to catch up.",
+				"duration", compactionDuration.String(),
+				"mint", mint,
+				"maxt", maxt,
+			)
+		}
```

Comment:
This should be after `db.head.truncateWAL`, and let's use `db.head.chunkRange` instead of `maxt-mint`.

Reply:
Wouldn't this be what I said before? Time spent on multiple compactions plus time spent on head Truncation?

Reply:
And does it make sense to log even if `returnErr != nil`? I think there's no need if an error is already being logged, right?

Reply:
Ignore this last comment, I was thinking about a deferred function for this log, but a deferred function would have to wait for db.compactBlocks() to finish, and that's not what we want